File: web-sources/s8-stack-web/carbide/objform/ObjFormLayer.js

import { CubeElement } from '/s8-stack-web/carbide/cube/CubeElement.js';
import { S8 } from '/s8-io-bohr/atom/S8.js';

S8.import_CSS("/s8-stack-web/carbide/dock/Dock.css");

/**
 * 
 */
export class ObjFormLayer extends CubeElement {

    /**
     * 
     */
    constructor(){
        super();

        this.node = document.createElement("div");
        this.node.classList.add("objform-box");
    }


    S8_render(){ /* continuous rendering approach... */ }

    /**
     * 
     * @param {*} root 
     */
    S8_set_root(root) {
       this.node.appendChild(root.getEnvelope());
    }


    getEnvelope() {
        return this.node;
    }

    S8_dispose(){ /* TODO */ }
}

File: sources/com/s8/stack/web/carbide/objform/ObjFormLayer.java
package com.s8.stack.web.carbide.objform;

import com.s8.io.bohr.neon.core.NeBranch;
import com.s8.stack.web.carbide.cube.CubeElement;


/**
 * 
 * @author pierreconvert
 *
 */
public class ObjFormLayer extends CubeElement {

	
	
	/**
	 * 
	 */
	public ObjFormLayer(NeBranch branch) {
		super(branch, "/s8-stack-web/carbide/objform/ObjFormLayer");
	}
	
	
	
	public void setRoot(ObjFormElement element) {
		vertex.setObj("root", element);
	}

	
}
